ui/src/contexts/APIProvider.tsx

```diff
@@ -23,6 +23,8 @@ interface APIProviderType {
   upstreamLogs: string;
   metrics: Metrics[];
   connectionStatus: ConnectionState;
+  versionInfo: VersionInfo | null;
```

Comment:
instead of `null` make versionInfo always VersionInfo, with a default value of "unknown" for build_date, commit and version. This will eliminate null checking in downstream code.